iml-api/src/graphql.rs

```diff
@@ -591,39 +635,100 @@ impl MutationRoot {
             .await
             .map_err(|e| e.into())
     }
-}
-
-async fn active_mgs_host_fqdn(
-    fsname: &str,
-    pool: &PgPool,
-) -> Result<Option<String>, iml_postgres::sqlx::Error> {
-    let fsnames = &[fsname.into()][..];
-    let maybe_active_mgs_host_id = sqlx::query!(
-        r#"
-            SELECT active_host_id from target WHERE filesystems @> $1 and name='MGS'
-        "#,
-        fsnames
-    )
-    .fetch_optional(pool)
-    .await?
-    .and_then(|x| x.active_host_id);
+    #[graphql(arguments(
+        fsname(description = "The filesystem to create snapshots with"),
+        interval(description = "How often a snapshot should be taken"),
+        use_barrier(
+            description = "Set write barrier before creating snapshot. The default value is `false`"
+        ),
+    ))]
+    /// Creates a new snapshot interval.
+    /// A recurring snapshot will be taken once the given `interval` expires for the given `fsname`.
+    /// In order for the snapshot to be successful, the filesystem must be available.
+    async fn create_snapshot_interval(
+        context: &Context,
+        fsname: String,
+        interval: GraphQLDuration,
+        use_barrier: Option<bool>,
+    ) -> juniper::FieldResult<bool> {
+        sqlx::query!(
+            r#"
+                INSERT INTO snapshot_interval (
+                    filesystem_name,
+                    use_barrier,
+                    interval
+                )
+                VALUES ($1, $2, $3)
+            "#,
+            fsname,
+            use_barrier.unwrap_or_default(),
+            PgInterval::try_from(interval.0)?,
+        )
+        .execute(&context.pg_pool)
+        .await?;
 
-    tracing::trace!("Maybe active MGS host id: {:?}", maybe_active_mgs_host_id);
+        Ok(true)
```

Comment:
Usually we are returning the `id` of the row inserted, so that then we can pass it to `remove_snapshot_interval`. Is the idiom not needed here?

Reply:
Typical graphql idiom is to return true if the mutation didn't have errors. It will be fairly atypical to remove the interval right after adding it. I suppose there could be some value to returning the entire created object, but I'd rather wait to see if that ends up being needed than implementing it now.